### PR TITLE
Token-level regression mode added in ForTokenClassification models

### PR DESCRIPTION
**Pull request by gsarti**
This is related to the issue #3646 I opened 2 days ago and which was considered interesting by @julien-c 

I added the **support for token-level regression in Bert, Roberta, Albert, XLNet, XLM, DistilBert and in the template for adding a new model** when `self.num_labels == 1`, fixing the docstring to match the new changes (and correcting the one for XLNetForTokenClassification which was copied from the XLNForMultipleChoice one).

Given two different approaches to compute `active_labels`, I privileged the one used in more recent models (e.g. Albert). The change was tested on tests and examples (without RUN_SLOW and RUN_CUSTOM_TOKENIZERS though) and passed all tests.

I didn't felt like adding this to Electra since its TokenClassification implementation seems to be missing the `num_labels` variable for some reason. I didn't add new tests since the sentence regression case wasn't covered by the testing suite either.

**Edit:** After the fix on `ElectraForTokenClassification`'s `num_labels` attribute, I added support for token-level regression there too.

### CODE REVIEW

**Comment by julien-c** [comment]
LGTM but I'll let others chime in.

**Comment by julien-c** [comment]
For ELECTRA, the discrepancy should have been fixed by 500aa12318ce5acd289d5edb6cb8266b3c3b162e, so can you propagate your changes there too?

**Comment by LysandreJik** [comment]
Related to naming, as it's for the `XXXForTokenClassification` when you have a single label wouldn't you expect to get a cross-entropy loss such as [binary cross-entropy](https://pytorch.org/docs/stable/nn.html#bceloss), rather than regression? Seeing as it's a classification model?

**Comment by gsarti** [comment]
I'd say the same would apply to `XXXForSentenceClassification`, right? I guess it would probably be best to decouple regression and classification for those two classes instead of having regression as a special case.

**Comment by BramVanroy** [comment]
Any chance of blowing new life into this? Token classification is particularly familiar due to NER, but in many research fields (e.g. psycholinguistic studies) we are interested in a lot more than that. Continuous values for tokens are very common there. I'd love to see regression and multilabel classification for token classification models.

**Comment by HoopsMcann** [comment]
@gsarti I'm interested in implementing a similar feature for my own work. I know the issue is stale now, but can you explain how this works with variable length token sequences (and thus different label lengths)?

**Comment by BramVanroy** [comment]
Sad to see this die :(
